net.solarnetwork.node.weather.ibm.wc/src/net/solarnetwork/node/weather/ibm/wc/BasicWCClient.java

```diff
@@ -107,33 +98,29 @@ private Integer parseInt(JsonNode n) {
 	//TODO valid time UTC
 	@Override
 	public Collection<GeneralDayDatum> readDailyForecast(String locationIdentifier, String apiKey,
-			String datumPeriod) {
+			DailyDatumPeriod datumPeriod) {
 
 		if ( locationIdentifier == null ) {
 			return null;
 		}
 		final List<GeneralDayDatum> result = new ArrayList<GeneralDayDatum>();
-		final String url = getURL(this.dailyForecastUrl, locationIdentifier, apiKey, datumPeriod);
+		final String url = getURL(this.dailyForecastUrl, locationIdentifier, apiKey,
+				datumPeriod.toString());
```

Comment:
I think it would be better to use the more explicit `datumPeriod.getPeriod()` method here.